presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/CommandExecutor.java

```diff
@@ -13,44 +13,79 @@
  */
 package com.teradata.prestomanager.agent;
 
+import com.google.common.io.ByteStreams;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-import java.io.BufferedReader;
-import java.io.InputStreamReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringTokenizer;
 import java.util.concurrent.TimeUnit;
-import java.util.stream.Collectors;
 
-import static java.lang.Runtime.getRuntime;
 import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
 
 public final class CommandExecutor
 {
     private static final Logger LOGGER = LogManager.getLogger(CommandExecutor.class);
+    private final String[] commandArray;
+    private int timeout;
 
-    private CommandExecutor() {}
+    private CommandExecutor(String[] commandArray)
+    {
+        this.commandArray = requireNonNull(commandArray);
+        timeout = 60;
+    }
+
+    public static CommandExecutor newCommandExecutor(String tokenizeCommand, String... args)
```

Comment:
I would just call this `commandExecutor`, or just `command`, but that's a matter of taste.